setup.py

```diff
@@ -30,11 +30,28 @@ def _minimal_ext_cmd(cmd):
         return out
 
     try:
-        out = _minimal_ext_cmd(['git', 'rev-parse', 'HEAD'])
-        git_revision = out.strip().decode('ascii')
+        out = _minimal_ext_cmd(cmd)
+        return out.strip().decode('ascii')
     except OSError:
-        git_revision = 'unknown-git'
-    return git_revision
+        pass
+
+
+def git_revision():
+    version = git_cmd(['git', 'rev-parse', 'HEAD'])
+    if not version:
+        version = 'unknown-git'
+    return version
+
+
+def git_version():
+    version = git_cmd(['git', 'describe', '--tag'])
```

Comment:
Running this code locally results in an `app/version.py` file:

```python
# THIS FILE IS GENERATED FROM SETUP.PY
version = 'v1.0.2.ce9aa231'
git_revision = 'ce9aa231a2c12df2a61b0569d628a09e480a344b'
full_version = '%(version)s.%(git_revision)s' % {
    'version': version,
    'git_revision': git_revision,
}
```

Let's strip the `v` prefix when loading the version from the code.  Ideally, we could also do the standard version output:

```python
import app
print(app.__version__)
```

to mirror the standard example like this:

```python
In [7]: import flask

In [8]: flask.__version__
Out[8]: '1.1.2'
```

Reply:
Yeah that's a good idea.  I'll change it

Reply:
I added `__version__` to both the flask app and the python `app` package:

```
In [1]: app
Out[1]: <Flask 'app'>

In [2]: app.__version__
Out[2]: '1.0.2.6a304eef'

In [3]: from app import __version__

In [4]: __version__
Out[4]: '1.0.2.6a304eef'
```